### PR TITLE
Add POST /detokenize and GET /props

### LINKED ISSUE #7: Implement /detokenize

### LINKED ISSUE #10: Implement /props

### PR DESCRIPTION
**Pull request by madrigal-cqv**
**Description**

- Add POST /detokenize
- Add GET /props

**Motivation**

- Implement core features

**Testing Done**
```
openapi-generator-cli validate -i ~/.../llamacpp-openapi/models/llamacpp.yml
No validation issues detected.
```

**Backwards Compatibility Criteria (if any)**

N/A

### CODE REVIEW

File: models/llamacpp.yml [hunk 1]
```diff
@@ -763,4 +791,31 @@ components:
...
+      type: object
+      properties:
+        content:
+          type: string
+          description: Text generated from tokens
+    GetPropsResponse:
+      type: object
+      properties:
+        system_prompt:
+          $ref: "#/components/schemas/SystemPrompt"
```
Comment (ProfHugo) [inline models/llamacpp.yml]: seems in this case system_prompt IS just a string; ie:
```
{
    "system_prompt": "The Assistant is helpful, kind, honest, good at writing, and never fails to answer the User's requests immediately and with precision.\\n\n",
    "default_generation_settings": { ... },
    "total_slots": 1,
    "chat_template": "..."
}
```
so we will need to change the type of this to a string

Comment (ProfHugo) [reply models/llamacpp.yml]: note that /completion and /infill still accepts the SystemPrompt object ie:
```
{
    "prompt":"Tell me something about yourself.",
    "system_prompt": {
        "prompt": "You are NOT a helpful assistant.\n",
        "anti_prompt": "User:",
        "assistant_name": "Assistant:"
    }
}
```

File: models/llamacpp.yml [hunk 2]
```diff
@@ -763,4 +791,31 @@ components:
...
+          type: array
+          items:
+            type: integer
+    PostDetokenizeResponse:
+      type: object
+      properties:
+        content:
+          type: string
+          description: Text generated from tokens
+    GetPropsResponse:
```
Comment (ProfHugo) [inline models/llamacpp.yml:808]: strange; seems like there's mismatch between what's on the documentation and the current state; this is the current structure on documentation
```
{
  "assistant_name": "",
  "user_name": "",
  "default_generation_settings": { ... },
  "total_slots": 1,
  "chat_template": ""
}
```
(+assistant_name, +user_name, -system_prompt)
whereas with both our containers it returns
```
{
    "system_prompt": "...",
    "default_generation_settings": { ... },
    "total_slots": 1,
    "chat_template": "..."
}
```
(-assistant_name, -user_name, +system_prompt)

Though I can't get a variant of the server to return something with assistant_name or user_name. I think the safe bet is to add support for those two string fields regardless.

**Review by ProfHugo — CHANGES_REQUESTED** [review]

File: models/llamacpp.yml (hunk 2 above)
Comment (madrigal-cqv) [reply models/llamacpp.yml:808]: added

File: models/llamacpp.yml (hunk 1 above)
Comment (madrigal-cqv) [reply models/llamacpp.yml]: fixed

**Review by ProfHugo — APPROVED** [review]
lgtm

**Pull request merged**